src/commonMain/kotlin/fr/acinq/lightning/channel/states/Offline.kt

```diff
@@ -99,15 +100,7 @@ data class Offline(val state: PersistedChannelState) : ChannelState() {
                     else -> Pair(Offline(newState), actions)
                 }
             }
-            cmd is ChannelCommand.Close.ForceClose -> {
-                val (newState, actions) = state.run { process(cmd) }
-                when (newState) {
-                    // NB: it doesn't make sense to try to send outgoing messages if we're offline.
-                    is Closing -> Pair(newState, actions.filterNot { it is ChannelAction.Message.Send })
-                    is Closed -> Pair(newState, actions.filterNot { it is ChannelAction.Message.Send })
-                    else -> Pair(Offline(newState as PersistedChannelState), actions)
-                }
-            }
+            cmd is ChannelCommand.Close.ForceClose -> state.run { handleLocalError(cmd, ForcedLocalCommit(channelId)) }
```

Comment:
It's probably fine to that we're not filtering the `error` message anymore, since we only handle `ChannelAction.Message.Send` if `_connectionState.value == Connection.ESTABLISHED`. One thing that surprises me though is that I don't see what will happen to this action in the disconnected case: the `when` in `processActions` (in `Peer.kt`) doesn't seem to be exhaustive...which is surprising because the compiler should complain about it?

Reply:
This may be due to the use of `when` without clause. Switch to `when` with clause in 6d057dca and made sure that it complains when pattern matching is not exhaustive.